eth/filters/filter.go

```diff
@@ -368,6 +366,16 @@ func (f *Filter) rangeLogs(ctx context.Context, firstBlock, lastBlock uint64) ([
 		if err := session.doSearchIteration(); err != nil {
```

Comment:
I think it's not necessary to return the logs while the error is not nil.

The result returned by `filter.Logs(ctx)` will be discarded if error is not nil
in `(api *FilterAPI) GetLogs`, `func (api *FilterAPI) GetFilterLogs` and in APIs of graphql

So, please simply discard all cached result if the error is not nil

Reply:
Ok, merged your changed.